controller/engine_image_controller.go

```diff
@@ -365,6 +365,17 @@ func (ic *EngineImageController) syncEngineImage(key string) (err error) {
 		// Allow update reference count and clean up even it's incompatible since engines may have been upgraded
 	}
 
+	if engineImage.Status.State != types.EngineImageStateIncompatible && !reflect.DeepEqual(types.GetEngineImageLabels(engineImage.Name), ds.Labels) {
```

Comment:
Add the link to the issue either in the comment here or in the commit message, so we can trackback to the reason we added this commit later, when we want to remove this workaround.